### PR TITLE
[RESTEASY-2204] spring-reactor RxInvoker (almost exact copy of rxjava2 one)

### PR DESCRIPTION
**Pull request by crankydillo**
[Project Reactor](https://projectreactor.io/)

I'm doing local builds of this code for one of my projects and wondered if you might want this.  Especially, since it's basically your code:)

### CODE REVIEW

**Comment by asoldano** [comment]
@crankydillo, interesting, we'll evaluate this, thanks. In the meantime, we require that all contributions are made under the terms of the ASL 2.0 License: http://www.apache.org/licenses/LICENSE-2.0 . Do you agree to these terms?"

**Comment by crankydillo** [comment]
Thanks.  Yes, the ASL 2.0 terms are fine.  This is basically your code anyway:)

File: resteasy-reactor/pom.xml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        <scope>test</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.jboss.resteasy</groupId>
+        <artifactId>resteasy-core-spi</artifactId>
+        <version>${project.version}</version>
+    </dependency>
+    <dependency>
+        <groupId>org.jboss.resteasy</groupId>
+        <artifactId>resteasy-context-propagation</artifactId>
```
Comment (crankydillo) [inline resteasy-reactor/pom.xml]: I don't think this is needed as I didn't do anything to support auto-context propagation.  Please let me know what I need to do here.

Comment (asoldano) [reply resteasy-reactor/pom.xml]: Considering the context propagation does not work, I think we could even remove this dependency. We'll need at least a very brief mention of this spring-reactor module in the documentation, and there we should also have a note on the context propagation being up to the user.

**Pull request merged**